src/pages/index.js

```diff
@@ -66,59 +91,202 @@ function handleImageClick(card) {
 }
 
 function handleAddFormSubmit(data) {
-  renderCard(data);
-  addPicPopup.close();
+  addPicPopup.renderLoading(true);
+  api
+    .addCard(data)
+    .then((data) => {
+      renderCard(data);
+      addPicPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addPicPopup.renderLoading(false);
+    });
 }
 
 function handleEditProfileFormSubmit(data) {
-  userInfo.setUserInfo(data);
-  // profileName.textContent = profileModalName.value;
-  // profileTitle.textContent = profileModalTitle.value;
+  editProfilePopup.renderLoading(true);
+
+  api
+    .editUserInfo(data)
+    .then(() => {
+      userInfo.setUserInfo(data);
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editProfilePopup.renderLoading(false);
+    });
+}
+
+let cardSection;
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([data, initialCards]) => {
+    console.log(data);
+    userInfo.setUserInfo(data);
+    profilePicture.src = data.avatar;
+    cardSection = new Section(
+      {
+        items: initialCards,
+        renderer: renderCard,
+      },
+      ".gallery__cards"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+function handleAvatarFormSubmit(data) {
+  updateAvatar.renderLoading(true);
   console.log(data);
-  editProfilePopup.close();
+  api
+    .updateProfilePic(data)
+    .then(() => {
+      console.log(data);
+      profilePicture.src = data.link;
+      updateAvatar.close();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      updateAvatar.renderLoading(false);
+    });
+}
+
+function handleAddLike(cardID) {
+  api
+    .addLike(cardID)
+    .then((data) => {
+      console.log(data);
+      this.setLikeState(data.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleRemoveLike(cardID) {
```

Comment:
**Could be improved**

To have access to any method of the class in `index.js` you can pass `this` to the call:

```js

        this._likeButton.addEventListener('click', () => {
            this._handleLike(this);
        });


```

And now in `index.js` you can call the methods like this:


```js


function addLike(cardInstance) {
        api.likeCard(cardInstance.getId())
            .then((res) => {
               cardInstance.addLike(res)
            })
            .catch((err) => {
                console.error(err);
            });
}


```

The same thing is with deleting cards